WordPress/Classes/ViewRelated/Media/Tenor/TenorResponse.swift

```diff
@@ -0,0 +1,166 @@
+import Foundation
+import MobileCoreServices
+
+typealias TenorGifFormat = TenorMedia.MediaCodingKeys
+
+class TenorResponse: NSObject, Decodable {
+    let next: String
+    let results: [TenorMedia]
+}
+
+class TenorMedia: NSObject, Decodable {
+    enum CodingKeys: String, CodingKey {
+        case id
+        case created
+        case itemurl
+        case title
+        case media
+    }
+
+    enum MediaCodingKeys: String, CodingKey, CaseIterable {
+        case nanogif
+        case tinygif
+        case mediumgif
+        case gif
+    }
+
+    let id: String
+    let created: Date
+    let itemurl: String
+    let title: String
+
+    // Data format in the response has unnecessary depth. We'll convert it to a more optimal format
+    var gifs: [TenorGifFormat: TenorGif]
+
+    required init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+
+        id = try container.decode(String.self, forKey: CodingKeys.id)
+
+        let createdDate = try container.decode(Float.self, forKey: CodingKeys.created)
+        created = Date(timeIntervalSince1970: Double(createdDate))
+
+        itemurl = try container.decode(String.self, forKey: CodingKeys.itemurl)
+        title = try container.decode(String.self, forKey: CodingKeys.title)
+
+        var mediaContainer = try container.nestedUnkeyedContainer(forKey: CodingKeys.media)
```

Comment:
Do you think we can move this to a method or add a comment to state what this logic is doing? In this case, we're iterating over all available media related to this gif, right?

Reply:
Yes. Most of this is boilerplate code well known to people familiar with the Codable protocol. The common practice here is to make the decoding in the init(from:) method (unless it is very complicated) because it is an initializer. I think adding a comment there would be a better solution if it isn't self-explanatory enough. What do you think?

Reply:
Yes, that what it's doing. I'll try to make it easier to understand.